includes/payment-methods/class-wc-stripe-upe-payment-method.php

```diff
@@ -642,7 +642,7 @@ public function payment_fields() {
 			</fieldset>
 			<?php
 			if ( $this->should_show_save_option() ) {
```

Comment:
Given the logic below inside the code block, this code is effectively a no-op when `! is_user_logged_in()`. So it's probably worth pulling that check up to this level so we're not calling any of the relevant filters unnecessarily.

This is technically not backwards-compatible, as code may have been triggering side-effects from those filters, but I doubt that's true for this specific filter.

**NOTE:** This is not blocking feedback -- I think we should be careful about making this change, and it definitely needs more thought and its own PR.
```suggestion
			if ( is_user_logged_in() && $this->should_show_save_option() ) {
```

Reply:
I would also feel more comfortable handling this refactor separately, as this may have other side effects.